### PR TITLE
feat: select refactor

### PR DESCRIPTION
**Pull request by rihansiddhi**
## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)

### CODE REVIEW

**Review by kaushal-freshworks — CHANGES_REQUESTED** [review]

File: packages/crayons-core/src/components/select/select.tsx [hunk 1]
```diff
@@ -23,6 +23,24 @@ import {
...
+  initializeComponent,
+  innerClickHandler,
+  innerFocusHandler,
+  innerOnBlur,
+  isValueEqual,
+  openDropdown,
+  optionsWatcher,
+  renderInput,
+  valueExists,
+} from '../../utils/select-utils';
```
Comment (kaushal-freshworks) [inline packages/crayons-core/src/components/select/select.tsx:43]: Now that there is no email select and select and have been merged, is the utils still needed ?

Comment (rihansiddhi) [reply packages/crayons-core/src/components/select/select.tsx:43]: This is not needed @kaushal-freshworks. Hence, I have closed this PR.

CC @arvindanta

File: packages/crayons-core/src/utils/select-utils.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+export function initializeComponent() {
+  //TODO: The below is a rough draft and needs to be optimized for better performance.
+  const selectOptions = Array.from(
+    this.host.querySelectorAll('fw-select-option')
+  );
+
+  // Set value if the selectedOptions is provided
+  if (this.selectedOptions.length > 0) {
+    this.selectedOptionsState = this.selectedOptions;
+    this.value = this.multiple
```
Comment (kaushal-freshworks) [inline packages/crayons-core/src/utils/select-utils.ts:48]: This is a bug ? Shouldnt this be this.value === this.multiple ?

File: packages/crayons-core/src/utils/select-utils.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+      ? this.selectedOptions.map((option) => option.value)
+      : this.selectedOptions[0].value;
```
Comment (kaushal-freshworks) [inline packages/crayons-core/src/utils/select-utils.ts:50]: I dont understand whats happening here. Please explain on a call .

File: packages/crayons-core/src/utils/select-utils.ts [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    this.value.length !== this.selectedOptions.length
+  ) {
+    this.selectedOptionsState = this.dataSource.filter((option) =>
+      this.isValueEqual(this.value, option)
+    );
+  }
+  if (this.dataSource?.length > 0) {
+    // Check whether the selected data in the this.dataSource  matches the value
+    const selectedDataSource = this.dataSource.filter(
+      (option) => option.selected
```
Comment (kaushal-freshworks) [inline packages/crayons-core/src/utils/select-utils.ts:88]: Array.filter is not the correct method to be used here as you are not checking for any condition but just returning the option.selected. Please use Array.forEach()

File: packages/crayons-core/src/utils/select-utils.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    );
+    const selectedDataSourceValues = selectedDataSource.map(
+      (option) => option.value
```
Comment (kaushal-freshworks) [inline packages/crayons-core/src/utils/select-utils.ts:91]: Dont know whats happening here. First filter() is done and on the same elements a map() is done. Looks like redundant loops running here. Please discuss this in a call. Why cant this be done in one single loop ? If the select has hundreds of items, these repeated loops will take forever to complete.

File: packages/crayons-core/src/utils/select-utils.ts [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+    ) {
+      this.value = selected;
+      this.selectedOptionsState = selectedDataSource;
+    }
+  }
+  this.host.addEventListener('focus', this.setFocus);
+
+  //Get id
+  this.hostId = this.host.id || '';
+}
```
Comment (kaushal-freshworks) [inline packages/crayons-core/src/utils/select-utils.ts:108]: This whole Init function looks like a big bundle of code. Am not able to clearly understand whats happening here. Please discuss this in a call.

**Pull request closed without merging**